### PR TITLE
Application bug fixes and performance enhancements

### LINKED ISSUE #18: Resolve network performance issues at Stanford

### LINKED ISSUE #42: Improve experiment loading performance using itk.js WorkerPool.
Using the WorkerPool, we can save the repeated loads of the heavy javascript (emscripten wrapped C++ library methods) associated with the imageio worker.  Instead of reloading those 3 heavy files for each image we need to read, we can just load them as many times as we have workers in the pool, which should provide much faster experiment loading.

This is under development in #39

### PR DESCRIPTION
**Pull request by scottwittenburg**
This PR fixes several issues with the application:

- Use itk.js WorkerPool to speed up loading
- Fix issues encountered when interrupting currently loading experiment
- Support export of original CSV format
- Work around issue using older python 3.5 until we can upgrade python in the docker image stack
- Fix load state indicators:
  * Previously scans were marked as loaded or not loaded using filled vs. empty disc next to each scan.  The computation for this was done efficiently, but was ultimately incorrect as the scan was marked as loaded as soon as a worker had picked up the last image to load.  In cases where the only image in a scan is really large, the scan could be marked as loaded before it was actually ready.
  * This change removes the filled circles indicating individual scans are loaded or not in favor of a single icon located next to the experiment name.

Closes #18. Closes #42.

### CODE REVIEW

**Comment by dzenanz** [comment]
This dragged in all the commits from #35, because it has not been merged yet.

**Comment by scottwittenburg** [comment]
Yes, thanks @dzenanz, if this becomes good enough to consider truly adding, I'll either tack it on to the end of that PR, or else rebase this on master (after that's merged).

**Comment by dzenanz** [comment]
Time to rebase :D

**Comment by scottwittenburg** [comment]
This will need the cancel api addition to the `WorkerPool` in this [PR](https://github.com/InsightSoftwareConsortium/itk-js/pull/395) to `itk.js` when it is merged and released.

File: client/src/components/JSONConfig.vue [hunk 1]
```diff
@@ -72,7 +72,10 @@ export default {
           name="miqa-json-exportpath"
           :rules="[
             v => !!v || 'path is required',
-            v => v.endsWith('.json') || 'Needs to be a json file'
+            v =>
+              v.endsWith('.json') ||
+              v.endsWith('.csv') ||
+              'Needs to be a json file'
```
Comment (scottwittenburg) [inline client/src/components/JSONConfig.vue]: I thought I had replaced this with "Needs to be a json or csv file", but I must have missed it.  I'll fix it before we merge.

**Review by dzenanz — APPROVED** [review]
Code looks good after a short look.

**Review by thewtex — APPROVED** [review]

**Comment by aashish24** [comment]
itk-js + miqa = big smile :)

File: devops/docker/Dockerfile [hunk 2]
```diff
@@ -9,7 +9,7 @@ RUN apt-get update && apt-get install -qy \
     libsasl2-dev && \
   apt-get clean && rm -rf /var/lib/apt/lists/*
 
-RUN wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
+RUN wget https://bootstrap.pypa.io/3.5/get-pip.py && python3 get-pip.py
```
Comment (jimklo) [inline devops/docker/Dockerfile]: This URL causes an error on build.

Should be `https://bootstrap.pypa.io/pip/3.5/get-pip.py` instead.

**Review by jimklo — CHANGES_REQUESTED** [review]
See comments.

**Comment by scottwittenburg** [comment]
@jeffbaumes This is the PR I was just mentioning

**Pull request merged**